src/evaluators/response/PHAL_ResponseSquaredL2DifferenceSide_Def.hpp

```diff
@@ -200,46 +196,52 @@ evaluateFields(typename Traits::EvalData workset)
             sum += sq * w_measure(sideSet_idx,qp);
           }
 
-          this->local_response_eval(cell, 0) = sum*scaling;
-          this->global_response_eval(0) += sum*scaling;
-        }
+          this->local_response_eval(cell,0) = sum*scaling;
+          KU::atomic_add<ExecutionSpace>(&(this->global_response_eval(0)), sum*scaling);
+        });
         break;
       case 1:
-        for (int sideSet_idx = 0; sideSet_idx < sideSet.size; ++sideSet_idx)
-        {
+        Kokkos::parallel_for(this->getName(),RangePolicy(0,sideSet.size),
+                       KOKKOS_CLASS_LAMBDA(const int& sideSet_idx) {
           // Get the local data of cell
-          const int cell = sideSet.ws_elem_idx.h_view(sideSet_idx);
+          const int cell = sideSet.ws_elem_idx.d_view(sideSet_idx);
+
+          ScalarT diff_1[8] = {0};
```

Comment:
Another case of an automatic tmp fad inside a kernel: are these ok? For SFad and SLFad, I think the answer is yes, but for generic Fad, I'm not sure. Someone with more Fad knowledge than me may know.

Reply:
That's a good point. With DFad it will create temporaries, and it could run out of memory.

Reply:
it's an issue with dfad on gpu but we don't use dfad.

Reply:
We do allow DFad during config though. If we don't want to allow DFad, we should make it clear and throw during config time.

Reply:
I'm not concerned about running out of memory. I'm more concerned with doing lots of small Cuda allocations at run time.

Reply:
yeah that makes sense. we should just set default slfad values (that work with testing) when running with cuda/hip/sycl instead of having those lines in every albany config. And error out if they are explicitly set to dfad.

when using sfad/slfad, they should be static allocations in local memory/registers. worse case, if the memory spills, it should be as performant as a read/write to global memory but not as bad as a device malloc, which is what's attempted with dfad (unless we setup the memory pool thing).

my concern would actually be, how large is the derivative dimension in this scalar? I forgot we're talking about optimization which could have a lot of derivative components... in which case, we may run out of memory...